### DISCUSSION TITLE
Specify the exact points of the begining and the ending of a given exchange

### DISCUSSION THREAD

**Discussion by yamus-didine** [strongswan/strongswan#1826] [Q&A]
Continuation of the discussion https://github.com/strongswan/strongswan/discussions/1809#discussion-5438673
So, I am interested in measuring the execution time of the exchanges `IKE_SA_INIT`, additional key exchange, and `CREATE_CHILD_SA`.
I am using `gettimeofday( )` function that allows measuring the elapsed time by calculating the difference between two `timeval` instances (for beginning and ending)
I am working on the file `task_manager_v2.c`
For now, I am able to perform time measurements between any two points of the code.
My inquiry is that I want to know what the defining points of the beginning and and the ending of a given exchange would be. In other terms, where exactly in the code I would place the `timeval` instances to record the time?

For example, in the file `task_manager_v2.c`, I placed the two 'timeval' instances as follows

![begin_time](https://github.com/strongswan/strongswan/assets/134303261/fe01283d-841d-46a5-9973-c6a7c99dae2d)

![end_time](https://github.com/strongswan/strongswan/assets/134303261/13d49f9e-8053-4fe2-9925-b19cd58469aa)

This was just a demonstration of what I did as an attempt to measure a task from one side (initiator or responder) from the activation moment until the transfer moment. Please, correct me if I am wrong.
In conclusion, I want to be able to measure the entire processing time of the aforementioned exchanges. I request guidance so I know what interval of time I should be recording.
